tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/ManagedIdentityTests.cs

```diff
@@ -37,6 +37,27 @@ public class ManagedIdentityTests : TestBase
         internal const string ExpectedErrorCode = "ErrorCode";
         internal const string ExpectedCorrelationId = "Some GUID";
 
+        [DataTestMethod]
+        [DataRow("http://127.0.0.1:41564/msi/token/", ManagedIdentitySource.AppService, ManagedIdentitySource.AppService)]
+        [DataRow(AppServiceEndpoint, ManagedIdentitySource.AppService, ManagedIdentitySource.AppService)]
+        [DataRow(ImdsEndpoint, ManagedIdentitySource.Imds, ManagedIdentitySource.Imds)]
+        [DataRow(null, ManagedIdentitySource.Imds, ManagedIdentitySource.Imds)]
+        [DataRow(AzureArcEndpoint, ManagedIdentitySource.AzureArc, ManagedIdentitySource.AzureArc)]
+        [DataRow(CloudShellEndpoint, ManagedIdentitySource.CloudShell, ManagedIdentitySource.CloudShell)]
+        [DataRow(ServiceFabricEndpoint, ManagedIdentitySource.ServiceFabric, ManagedIdentitySource.ServiceFabric)]
+        public void GetManagedIdentityTests(
```

Comment:
```suggestion
        public void GetManagedIdentityTests(
    string endpoint,
    ManagedIdentitySource expectedManagedIdentitySource)
```

you just need one and then compare it with actual

Reply:
The only difference is for IMDS so passing the expected one to